CMakeLists.txt

```
@@ -62,6 +62,20 @@ if(JWT_EXTERNAL_PICOJSON)
   find_package(picojson 1.3.0 REQUIRED)
 endif()
 
+find_package(nlohmann_json CONFIG)
+
+if(NOT nlohmann_json_FOUND)
+  include(FetchContent)
+
+  FetchContent_Declare(
+    json
```

Probably worth using the project name?  https://github.com/nlohmann/json/blob/a3e6e26dc83a726b292f5be0492fcc408663ce55/CMakeLists.txt#L7 I suspect we will be adding these over time

https://github.com/search?l=CMake&q=FetchContent_Declare+nlohmann_json&type=Code vs https://github.com/search?l=CMake&q=FetchContent_Declare+_json&type=Code  shows theres a wide range then I expected

```suggestion
    nlohmann_json
```